tidy3d/components/simulation.py

```diff
@@ -3297,8 +3398,9 @@ def snap_to_grid(geom: Geometry, axis: Axis) -> Geometry:
                 temp_structure = structure.updated_copy(geometry=temp_geometry, medium=new_medium)
 
                 if structure.medium.is_pec:
-                    pec_delta = fp_eps * max(np.abs(snapped_center), 1.0)
-                    new_bounds = (snapped_center - pec_delta, snapped_center + pec_delta)
+                    pec_plus = np.nextafter(snapped_center, +np.inf, dtype=np.float32)
+                    pec_minus = np.nextafter(snapped_center, -np.inf, dtype=np.float32)
+                    new_bounds = (pec_minus, pec_plus)
```

Comment:
It's not equivalent, but it is a more robust method to add a small increment to a floating point number, which I think is the goal of this step. Simply adding fp_eps is not correct, since fp_eps is the machine precision with respect to 1.0. Adding fp_eps to a 32bit float larger than 1 will not change it.

Maybe I misinterpreted the purpose of this step, but if you want the smallest increment for a given float size then the new changes are preferred.

Reply:
I saw this usage in a couple of different places and I had an initial quick fix, but I later found the np.nextafter function which is better.